compiler/src/dmd/mustuse.d

```diff
@@ -104,6 +107,25 @@ private bool isAssignment(Expression e)
     return false;
 }
 
+/**
+ * Returns: true if `e` is
+ *     `(this.field = typeof(this.field).init, this.field.__ctor(arg))`
+ */
```

Comment:
```suggestion
/**
 * Returns: true if `e` is an assignment that's been rewritten to a
 * constructor call.
 */
```
There is more than one rewrite to check for here; see comment on test case below.